js/game_board.js

```diff
@@ -0,0 +1,247 @@
+import _ from 'lodash';
+import d3 from 'd3';
+import stringify from 'json-stringify-pretty-compact';
+
+import {animationStepDurationMin, animationStepDurationMax} from './config';
+import {Stock} from './stock';
+import {Level} from './level';
+import {BareBoard} from './bare_board';
+
+
+// FIX level loading/storing is still too hackish
+// TODO decide where to use winning status; it seems I should move it here
+// TODO tob_bar needs a separate module
+
+export class GameBoard {
+  constructor(level, svg, helper, titleManager, levels, progressPearls, storage) {
+
+    this.bareBoard = new BareBoard(svg, {
+      experimentDisturbed: this.experimentDisturbedCallback.bind(this),
+      tileRotated: this.tileRotatedCallback.bind(this),
+      tileMouseover: this.tileMouseoverCallback.bind(this),
+      animationStart: this.animationStartCallback.bind(this),
+      animationEnd: this.animationEndCallback.bind(this),
+    });
+
+    this.levels = levels;
+    this.levelsLookup = _.indexBy(levels, (levelRecipe) => `${levelRecipe.group} ${levelRecipe.name}`);
+
+    this.helper = helper;
+    this.titleManager = titleManager;
+    this.progressPearls = progressPearls;
+    this.storage = storage;
+
+    this.stock = new Stock(svg, this.bareBoard);
+    this.bareBoard.stock = this.stock;  // such monkey patching not nice
+    this.logger = this.bareBoard.logger;
+    this.logger.logAction('initialLevel');
+
+    this.loadLevel(level);
+  }
+
+  experimentDisturbedCallback() {
+    this.titleManager.displayMessage(
+      'Experiment disturbed! Quantum states are fragile...',
+      'failure');
+  }
+
+  tileRotatedCallback(d) {
+    this.showTileHelper(d);
+  }
+
+  tileMouseoverCallback(d) {
+    this.showTileHelper(d);
+  }
+
+  animationStartCallback() {
+    this.titleManager.displayMessage(
+      'Experiment in progress...',
+      'progress');
+  }
+
+  animationEndCallback() {
```

Comment:
saveProgress - pisałem, niedopatrzenie

particleAnimation - to trzeba było zmienić (w ten czy inny sposób), bo przy obecnej architekturze by się nie rodzieliło sensownie boarda; to był znacznie cięższy refactoring niż kopiuj wklej do dwóch plików

konsole log - niedopatrzenie